src/components/Path/index.js

```diff
@@ -0,0 +1,85 @@
+import React, { useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import RenderIf from '../RenderIf';
+import CheckMark from './icons/checkMark';
+import Exclamation from './icons/exclamation';
+import { StyledContainer, StyledStepsList, StyledStepItem } from './styled';
+import { isStepSelected, getActiveStepIndex } from './helpers';
+
+export default function Path(props) {
+    const { currentStepName, onClick, children, id, className, style } = props;
+    const [hoveredStepName, setHoveredStepName] = useState(null);
+
+    const steps = useMemo(() => {
+        const selectedIndex = children.findIndex(child => child.props.name === currentStepName);
+        const hoveredIndex = children.findIndex(child => child.props.name === hoveredStepName);
+        const someStepHasError = children.some(child => child.props.hasError);
+
+        return React.Children.map(children, (child, index) => {
```

Comment:
we should use context for this since with this approach we can do some things, for example, if we want to use a RenderIf to show some PathStep dynamically, this can't be done because RenderIf will be the child, this is a drawback in our table component right now